app/controllers/providers/other_assets_controller.rb

```diff
@@ -8,7 +8,8 @@ def show
     def update
       authorize @legal_aid_application
       @form = Citizens::OtherAssetsForm.new(form_params)
-      render :show unless save_continue_or_draft(@form)
+      @form.errors.add :base, :provider_none_selected unless draft_selected? || none_checkbox_selected? || @form.any_checkbox_checked?
```

Comment:
If you look at the latest versions of other controllers using the same logic, you'll see that the sequence needs to be: `none_checkbox_selected? || @form.any_checkbox_checked? || draft_selected?` or as Julien pointed out, selecting "Save as draft" won't save the selected choices.